src/apps/chifra/internal/names/handle_terms.go

```diff
@@ -2,37 +2,93 @@ package namesPkg
 
 import (
 	"context"
+	"errors"
+	"io"
 	"os"
 
 	"github.com/TrueBlocks/trueblocks-core/src/apps/chifra/pkg/logger"
 	"github.com/TrueBlocks/trueblocks-core/src/apps/chifra/pkg/names"
 	"github.com/TrueBlocks/trueblocks-core/src/apps/chifra/pkg/output"
 	"github.com/TrueBlocks/trueblocks-core/src/apps/chifra/pkg/types"
+	"github.com/TrueBlocks/trueblocks-core/src/apps/chifra/proto"
 )
 
 func (opts *NamesOptions) HandleTerms() error {
-	namesArray, err := names.LoadNamesArray(opts.Globals.Chain, opts.getType(), names.SortByAddress, opts.Terms)
-	if err != nil {
-		return err
-	}
-	if len(namesArray) == 0 {
-		logger.Warn("No results for", os.Args)
-		return nil
-	}
+	var fetchData func(modelChan chan types.Modeler[types.RawName], errorChan chan error)
 
-	ctx := context.Background()
+	// Try RPC
+	grpcCtx, grpcCancel := proto.GetContext()
+	defer grpcCancel()
+	conn, client, grpcErr := proto.Connect(grpcCtx)
+	if grpcErr == nil {
+		defer conn.Close()
+		// RPC server is running and available
+		fetchData = func(modelChan chan types.Modeler[types.RawName], errorChan chan error) {
+			opts.fetchFromGrpc(client, modelChan, errorChan)
+		}
+	} else {
+		if !errors.Is(grpcErr, context.DeadlineExceeded) {
+			logger.Error("gRPC connection error:", grpcErr)
+		}
 
-	// Note: Make sure to add an entry to enabledForCmd in src/apps/chifra/pkg/output/helpers.go
-	fetchData := func(modelChan chan types.Modeler[types.RawName], errorChan chan error) {
-		for _, name := range namesArray {
-			name := name
-			modelChan <- &name
+		namesArray, err := names.LoadNamesArray(opts.Globals.Chain, opts.getType(), names.SortByAddress, opts.Terms)
+		if err != nil {
+			return err
+		}
+		if len(namesArray) == 0 {
```

Comment:
Do we handle the `no results` case in the RPC server path?

Reply:
Added, thanks 👍